analyzer/src/main/resources/scripts/bundler_dependencies_metadata.rb

```diff
@@ -0,0 +1,40 @@
+#!/usr/bin/ruby
+
+# Copyright (C) 2020 Bosch.IO GmbH
+# Copyright (C) 2017-2019 HERE Europe B.V.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# SPDX-License-Identifier: Apache-2.0
+# License-Filename: LICENSE
+#
+# This script mimics the behavior of calling the CLI command `bundle lock`, which
+# resolves the dependencies inside the Gemfile and writes them together with the
+# respective version to a lock file. Internally, Bundler tries to find the gemspec
+# files of the dependencies, both locally and remotely, and retrieves the respective
+# metadata. However, except for the name, version, and transitive dependencies, Bundler
+# discards the rest of the metadata. Thus, this script follows the steps but prints
+# the metadata to be further processed by other tools.
+# See https://github.com/rubygems/rubygems/blob/master/bundler/lib/bundler/cli/lock.rb
+
+require 'bundler'
+
+definition = Bundler.definition()
+
+definition.resolve_remotely!
+
+definition.specs.each do |spec|
+  next unless spec.name != "bundler"
+  puts "\n-\n"
```

Comment:
Is this separator arbitrarily chosen, or is there a deeper meaning? Consider adding a code comment.

Reply:
Yes it is chosen arbitrarily, but there is some problem with Windows, probably because a line break is `\r\n` in this context.
Do you have a better idea? The reason I chose a separator, is that I want to extract the single yamls since parsing the whole output did not work (Only the last yaml entry was parsed in the end). Another problem I experienced is that meta data can contain license text which might contain characters that interfere with the separator.

Reply:
So how about using "\0" (a `null` character) instead?